fio-ioengine/flexalloc.c

```diff
@@ -280,11 +335,20 @@ static int fio_flexalloc_init(struct thread_data *td)
 	fad->object = calloc(td->o.nr_files, sizeof(*fad->object));
 	assert(fad->object);
 
-	pthread_mutex_lock(&fa_mutex);
-	ret = fio_flexalloc_open(o->dev_uri, o->md_dev_uri, &fad->fs, td->thread_number);
-	if (ret) {
-		log_err("flexalloc: unable to open file system\n");
-		goto done;
+	if (daemon_mode(o))
+	{
+		ret = fio_flexalloc_open_daemon(fad, o);
+		if (ret) {
+			log_err("flexalloc: unable to open file system\n");
+			goto done;
+		}
+	} else {
+		pthread_mutex_lock(&fa_mutex);
+		ret = fio_flexalloc_open_direct(o->dev_uri, o->md_dev_uri, td->thread_number, &fad->fs);
+		if (ret) {
+			log_err("flexalloc: unable to open file system\n");
+			goto done;
+		}
```

Comment:
move the identical if (ret) { log_err() ... } lines outside of if (daemon_mode(o)) {} else {}